### PR TITLE
[WIP] Integrate React Piechart panel

### PR DESCRIPTION
**Pull request by corpglory-dev**
PR adds:

* Piechart component in `@grafana/ui`
* React Piechart panel

- [x] Rendering
- [x] Tooltip
- [ ] Legend

### CODE REVIEW

**Comment by torkelo** [comment]
would be interesting to define the interface for this component:

```ts
export interface Props extends Themeable {
  height: number;
  width: number;

  unit: string;
  value: number;
  pieType: string;
  format: string;
  stat: string;
  strokeWidth: number;
}
```

this will obviously not work as it will need many "series" or rows. Question is should it take  TimeSeriesVMs and some Table model format. Or some higher level abstraction that the PieChartPanelPlugin will have to transform data into. 

like 

```
export interface PieChartDataPoint {
  value: number;
  name: string; 
  color: string;
}

export interface Props extends Themeable {
  height: number;
  width: number;
  unit: string;  
  datapoints: PieChartDataPoint[]  
}
```

File: public/app/plugins/panel/piechart/plugin.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "type": "panel",
+  "name": "PieChart",
+  "id": "piechart",
+
```
Comment (torkelo) [inline public/app/plugins/panel/piechart/plugin.json]: can we add a "state": "alpha"  here? Then we can merge this to master without it being "finished" . To be able to use alpha panels you need to modify you ini file to include:

[panels]
enable_alpha = true

**Comment by torkelo** [comment]
This looking like a good start! Think we could merge this soon and continue refining it other PRs. 

Please see my comment on adding a alpha state to it.

**Comment by torkelo** [comment]
Can you try adding a simple storybook story for the piechart ? that would nice.  See this for reference:
https://github.com/grafana/grafana/pull/15466/files#diff-7d3f4c399bb414d9c6a2b4e3f08b2c85

File: packages/grafana-ui/src/components/Piechart/Piechart.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
+import React, { PureComponent } from 'react';
+import * as d3 from 'd3';
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Piechart/Piechart.tsx]: can you import only what your using here?

File: packages/grafana-ui/src/components/Piechart/Piechart.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+import { GrafanaThemeType } from '../../types';
+import { Themeable } from '../../index';
+
+export enum PiechartType {
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Piechart/Piechart.tsx]: Rename Piechart to PieChart

**Review by peterholmberg — CHANGES_REQUESTED** [review]
Since pie chart is two words, I'd like to use PieChart instead of Piechart everywhere in this PR

**Comment by corpglory-dev** [comment]
@torkelo @peterholmberg thanks for the review, code is fixed

**Comment by torkelo** [comment]
don't work on legend for this new piechart yet. think we can reuse a common legend component for the new graph panel hopefully

**Pull request merged**